### DISCUSSION TITLE
Add attributes to component script

### DISCUSSION THREAD

**Discussion by djgamerr** [GrapesJS/grapesjs#4731] [Q&A] [answered]
As of right now, if i load a already created website with grapesjs with allowScripts: true, it will also copy the <script> that the component has previously exported.

I have no way of recognizing the old script and removing it before the component adds the new script.
If i were to edit a single page 10 times, i would have 10-11 exactly the same scripts, which is not desireable.
Of course, there are other custom scripts, so I can't just remove all script tags.

Is there a way to add an attribute to components script so that I can detect it and remove it before i load the HTML?
In your documentation, it can be a string or an function.
How about an object {fn: ..., attributes: {}} just like components?

This way we could do something like this:
{fn:  () => null, attributes: {defer: true, "data-id": "some-id"}}

Thanks in advance.

**Comment by artf** [comment, marked as answer]
Use the JSON to load properly the project back not the output HTML, that's not the only issue you might face by importing an exported HTML.

**Comment by djgamerr** [comment]
Thank you
I will try doing that
